website/content/docs/k8s/dns/views/enable.mdx

```diff
@@ -0,0 +1,115 @@
+    Learn how to schedule a Consul DNS proxy for a Kubernetes Pod so that your services can return Consul DNS results for service discovery.
+---
+
+# Enable Consul DNS proxy for Kubernetes
+
+This page describes the process to deploy a Consul DNS proxy in a Kubernetes Pod so that Services can resolve Consul DNS requests. For more information, refer to [Consul DNS views for Kubernetes](/consul/docs/k8s/dns/views).
+
+## Prerequisites
+
+You must meet the following minimum application versions to enable the Consul DNS proxy for Kubernetes:
+
+- Consul v1.20.0 or higher
+- Either Consul on Kubernetes or the Consul Helm chart, v1.6.0 or higher
+
+## Update Helm values
+
+To enable the Consul DNS proxy, add the required [Helm values](/consul/docs/k8s/helm) to your Consul on Kubernetes deployment:
+
+```yaml
+connectInject:
+  enabled: true
+dns:
+  enabled: true
+  proxy: true
+```
+
+### ACLs
+
+We recommend you create a dedicated [ACL token with DNS permissions](/consul/docs/security/acl/tokens/create/create-a-dns-token) for the Consul DNS proxy. The Consul DNS proxy requires the following ACL permissions:
+
+```hcl
+node_prefix "" {
+  policy = "read"
+}
+
+service_prefix "" {
+  policy = "read"
+}
+```
+
+You can manage ACL tokens with Consul on Kubernetes, or you can configure the DNS proxy to access a token stored in Kubernetes secret. To use a Kubernetes secret, add the following configuration to your Helm chart:
+
+```yaml
+dns:
+  proxy:
+    aclToken:
+      secretName: <Consul-DNS-Token>
+      secretKey: <Token-Value>
+```
+
+## Retrieve Consul DNS proxy's address
+
+To look up the IP address for the Consul DNS proxy in the Kubernetes Pod, run the following command:
+
+```shell-session
+$ kubectl get services –-all-namespaces
+
+NAMESPACE     NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                  AGE
+consul        consul-connect-injector   ClusterIP   10.96.41.171   <none>        443/TCP                  12m
+consul        consul-dns                ClusterIP   10.96.148.46   <none>        53/TCP,53/UDP            12m
+default       kubernetes                ClusterIP   10.96.0.1      <none>        443/TCP                  21m
+kube-system   kube-dns                  ClusterIP   10.96.0.10     <none>        53/UDP,53/TCP,9153/TCP   21m
+```
+
+Note the `CLUSTER-IP` for the `consul-dns` service. Use this address when you update the ConfigMap resource.
+
+## Update Kubernetes ConfigMap
+
+Create or update a [ConfigMap object in the Kubernetes cluster](https://kubernetes.io/docs/concepts/configuration/configmap/) so that Kubernetes forwards DNS requests with the `.consul` domain to the IP address of the Consul DNS proxy.
+
+The following example of a ConfigMap resource configures Kubernetes to forward Consul DNS requests in the cluster to the Consul DNS Proxy running on `10.150.0.1`.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns
+  namespace: kube-system
+data:
+  Corefile: |
```

Comment:
Several Kubernetes distributions support using a `coredns-custom` config map to modify the CoreDNS configuration without modifying the original `Corefile`. For example,

```yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns-custom
  namespace: kube-system
data:
  consul.server: |
    consul:53 {
      errors
      cache 30
      forward . <consul-dns-proxy-from-above>
      reload
    }
```

<https://learn.microsoft.com/en-us/azure/aks/coredns-custom>

Should the docs prefer this config method with a note that some users may need to include the customizations in the original `coredns` config map?